CairoMakie/src/infrastructure.jl

```diff
@@ -240,6 +250,43 @@ function draw_plot(scene::Scene, screen::CairoScreen, primitive::Combined)
     return
 end
 
+# Possible improvements for this function:
+# - Obtain the bbox of the plot and draw an image which tightly fits that bbox
+#   instead of the whole Scene
+# - Recognize when a screen is an image surface, and set scale to render the plot
+#   at the scale of the device pixel
+function draw_plot_as_image(scene::Scene, screen::CairoScreen, primitive::Combined, scale = 1)
+    # you can provide `p.rasterize = scale::Int` or `p.rasterize = true`,
+    # if true then scale = 1.
+    scale == true && (scale = 1)
+    @assert scale isa Int || scale isa Bool
+    # Extract scene width in pixels
+    w, h = Int.(scene.px_area[].widths)
+    # Create a new Screen which renders directly to an image surface,
+    # specifically for the plot's parent scene.
+    scr = CairoScreen(scene; device_scaling_factor = scale)
+    # Draw the plot to the screen, in the normal way
+    draw_plot(scene, scr, primitive)
+
+    # Now, we draw the rasterized plot to the main screen.
+    # Since it has already been prepared by `prepare_for_scene`,
+    # we can draw directly to the Screen.
+    Cairo.rectangle(screen.context, 0, 0, w, h)
+    Cairo.save(screen.context)
+    Cairo.translate(screen.context, 0, 0)
+    # Cairo.scale(screen.context, w / scr.surface.width, h / scr.surface.height)
+    Cairo.set_source_surface(screen.context, scr.surface, 0, 0)
+    p = Cairo.get_source(scr.context)
+    # this is needed to avoid blurry edges
+    Cairo.pattern_set_extend(p, Cairo.EXTEND_PAD)
+    # Set filter doesn't work!?
+    Cairo.pattern_set_filter(p, Cairo.FILTER_BEST)
```

Comment:
Vector backends don't support that filter, see: https://github.com/JuliaPlots/Makie.jl/pull/1828
So, we should remove it?